Implemented download_pdb_in_range()

EricFlorin opened the pull request:
`download_pdb_in_range()` downloads and writes PDB files, whose PDB IDs are within a given range [lower, upper].

`download_pdb_in_range()` takes three inputs: `lower`, `upper`, and `dir`:
- `lower` and `upper` are strings that represents the range of IDs to look for PDB files. The range is from `lower` to `upper` inclusive. `lower` and `upper` can take in IDs with the wildcard character '*'; for example, if a user defines a range as "1A*" to "2*", then the function would attempt to download all PDB files with IDs between 1A00 to 2000 inclusive.
- `dir` is a string that represents a directory for the PDB files to be written to.

EricFlorin committed:
Started work on download_all_existing_pdb()

This function will download all PDB files from the Protein Bank Database. This will be done by looking through all possible protein IDs, and downloading the ones that exist in the search.

EricFlorin committed:
Implemented download_pdb_in_range()

This function in PDBGenerator would download all existing PDB files in the range of [lower, upper], where "lower" and "upper" are the lower and upper bound IDs. The PDB files are stored in the director given by "dir".

chuckie82 commented on DeepProjection/pdbGenerator.py:
```diff
@@ -36,36 +36,176 @@ def get_random_pdb(self):
     
-    def download_all_existing_pdb(self):
-        # Downloads all PDB files from the Protein Data Bank
+    def download_pdb_in_range(self, lower, upper, dir):
```
May be useful to assign default values: upper=None and dir=None
If upper is None, then just use lower to determine what to download.
If dir is None, download in current working directory

chuckie82 commented on DeepProjection/pdbGenerator.py:
```diff
@@ -36,36 +36,176 @@ def get_random_pdb(self):
+        # If lower == upper, our loop code should be able to handle that case on its own.
+
+        if lower == '*' and upper == '*': # Special Range 2: lower == '*' and upper == '*', so we will download all files
+            lower = '0000'
```
First character in the PDB ID is between 1 and 9. Not zero.
In other characters, zero to 9.

chuckie82 commented on DeepProjection/pdbGenerator.py:
```diff
@@ -36,36 +36,176 @@ def get_random_pdb(self):
 
-        # For debug
-        print(len(list_of_valid_ids))
+                                print("downloaded: ", currentID)
```
remove print statements

chuckie82 commented on DeepProjection/pdbGenerator.py:
```diff
@@ -36,36 +36,176 @@ def get_random_pdb(self):
+        dir : str
+            The directory to store the PDB files in.
+
+        Corner Cases
```
good job thinking about corner cases.
Add these tests to the /tests folder and run pytest and see if they pass

chuckie82 reviewed (commented):
Adding tests for the new function will improve codecov coverage value. Let's keep it above 90%

EricFlorin committed:
Added tests in test_pdbGenerator.py for download_pdb_in_range() and updated download_pdb_in_range()

Updated download_pdb_in_range() to include the changes mentioned in pull request #1 for chuckie82/DeepProjection.

Added tests for download_pdb_in_range() in test_pdbGenerator.py. The tests found some bugs in the input check portion of the function, which were then dealt with.

chuckie82 commented on DeepProjection/tests/test_pdbGenerator.py:
```diff
@@ -13,4 +13,182 @@ def test_getting_pdb(empty_pdbgenerator):
+        1. If upper == None, then range is from [lower, 9ZZZ].
+        2. If dir == None, then download PDB files to current working directory.
+    """
+    # There is no PDB with an ID of '9zzz', so it will be used in default value test.
```
9zzz may appear one day and this would fail. I think a better check would be a pdb starting with zero, i.e. 0aaa

chuckie82 commented on DeepProjection/tests/test_pdbGenerator.py:
```diff
@@ -13,4 +13,182 @@ def test_getting_pdb(empty_pdbgenerator):
+    assert empty_pdbgenerator.download_pdb_in_range('7Zz0', '0Zz1', 'FAKEDIR') == False
+
+    assert empty_pdbgenerator.download_pdb_in_range('0AAY', '7AAZ', None) == False
+    assert empty_pdbgenerator.download_pdb_in_range('0AAY', '7AAZ', 'FAKEDIR') == False
```
there are too many tests here. Once you test for upper and lower cases, there's no need to keep on testing this
Same with 'FAKEDIR' test

let's only do 1 or 2 per input error type.

chuckie82 commented:
Looking at pypdb code, there's seems to be an API for getting the scores as well:
https://github.com/williamgilpin/pypdb/blob/8bc4aa3b12cc5c0b5e6e7ee6f51b20875324e04c/pypdb/clients/search/search_client.py#L199

EricFlorin committed:
Trimmed the number of download_pdb_in_range() tests
